### PR TITLE
#582: Gracefuly handle missing payment provider credentials

### LINKED ISSUE #582: Disable webhook handlers when given payment provider is not enabled
# Original Sentry Issue:

TypeError: Missing parameter "apiKey" or "accessToken".

Sentry Issue: [SALEOR-APP-CHECKOUT-6](https://sentry.io/organizations/saleor/issues/3693594703/?referrer=github_integration)

```
TypeError: Missing parameter "apiKey" or "accessToken".
  File "./backend/payments/providers/mollie/utils.ts", line 11, in getMollieClient
    return createMollieClient({
  File "./backend/payments/providers/mollie/webhookHandler.ts", line 13, in verifyPayment
    const mollieClient = await getMollieClient();
  File "./utils/promises.ts", line 19, in unpackPromise
    const result = await promise;
  File "./pages/api/webhooks/mollie.ts", line 18, in handler
    const [paymentError, paymentData] = await unpackPromise(verifyPayment(id));
...
(5 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by mmiszy**
Fixes #582

### CODE REVIEW

**Review by bmigirl — APPROVED** [review]

File: apps/saleor-app-checkout/pages/api/webhooks/mollie.ts [hunk 1]
```diff
@@ -31,6 +32,12 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
 
   if (paymentError) {
     console.error(paymentError);
+
+    if (paymentError instanceof MissingPaymentProviderSettingsError) {
+      res.status(500).json({ error: paymentError.message });
```
Comment (witoszekdev) [inline apps/saleor-app-checkout/pages/api/webhooks/mollie.ts:37]: Why are we using

`res.status(500)` here and

```js
return Response.InternalServerError({ error: error.message });
```

in other places? 🤔

**Review by witoszekdev — COMMENTED** [review]
This is nice - app won't crash any more, but maybe we should do more?
If payment gateway isn't configured, maybe it should be disabled in dashboard? 🤔

File: apps/saleor-app-checkout/backend/payments/errors.ts [hunk 2]
```diff
@@ -26,3 +26,11 @@ export class UnknownPaymentError extends Error {
     Object.setPrototypeOf(this, UnknownPaymentError.prototype);
   }
 }
+
+export class MissingPaymentProviderSettingsError extends Error {
+  constructor(public provider: PaymentProviderID, missingOptions: readonly string[]) {
+    const opts = missingOptions.join(", ");
+    super(`Error! Provider: ${provider} | Missing payment provider configuration: ${opts}`);
+    Object.setPrototypeOf(this, MissingPaymentProviderSettingsError.prototype);
```
[the remark this answers is not in the document]
Comment (mmiszy) [reply apps/saleor-app-checkout/backend/payments/errors.ts:34]: This is required when extending built in classes in TypeScript.

File: apps/saleor-app-checkout/pages/api/webhooks/mollie.ts (hunk 1 above)
Comment (mmiszy) [reply apps/saleor-app-checkout/pages/api/webhooks/mollie.ts:37]: Because in some places we're using retes.

**Comment by mmiszy** [comment]
> If payment gateway isn't configured, maybe it should be disabled in dashboard? 🤔
Not part of this ticket but it's on the roadmap.

**Pull request merged**